File: app/src/main/java/com/coinninja/coinkeeper/service/client/model/CNSharedMemo.java
package com.coinninja.coinkeeper.service.client.model;

public class CNSharedMemo {

    Integer encrypted_format;
    String address;
    String encrypted_payload;
    String txid;

    public Integer getEncrypted_format() {
        return encrypted_format;
    }

    public void setEncrypted_format(Integer encrypted_format) {
        this.encrypted_format = encrypted_format;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getEncrypted_payload() {
        return encrypted_payload;
    }

    public void setEncrypted_payload(String encrypted_payload) {
        this.encrypted_payload = encrypted_payload;
    }

    public String getTxid() {
        return txid;
    }

    public void setTxid(String txid) {
        this.txid = txid;
    }

}

File: app/src/main/java/com/coinninja/coinkeeper/service/client/model/CNSharedMemo.kt
package com.coinninja.coinkeeper.service.client.model

data class CNSharedMemo(
        var txid: String? = null,
        var address: String? = null,
        var identity_hash: String? = null,
        var encrypted_payload: String? = null,
        var encrypted_format: String? = "2"
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as CNSharedMemo

        if (txid != other.txid) return false
        if (address != other.address) return false
        if (identity_hash != other.identity_hash) return false
        if (encrypted_payload != other.encrypted_payload) return false
        if (encrypted_format != other.encrypted_format) return false

        return true
    }

    override fun hashCode(): Int {
        var result = txid?.hashCode() ?: 0
        result = 31 * result + (address?.hashCode() ?: 0)
        result = 31 * result + (identity_hash?.hashCode() ?: 0)
        result = 31 * result + (encrypted_payload?.hashCode() ?: 0)
        result = 31 * result + (encrypted_format?.hashCode() ?: 0)
        return result
    }
}
